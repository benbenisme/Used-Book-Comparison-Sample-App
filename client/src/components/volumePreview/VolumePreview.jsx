import React from 'react';
import {
    Card,
    CardBody,
    CardTitle,
    CardSubtitle,
    CardText,
    Container,
    Row,
    Col
} from 'reactstrap';
import {
    Link
} from "react-router-dom";
import { useStores } from '../../state/index';
// import "./volumePreview.css"


export default function     VolumePreview(props) {
    const { volumeStore } = useStores();

    const volumeTitle = props.volumeResponse.volumeInfo.title;
    const volumeDescription = props.volumeResponse.volumeInfo.description?.substring(0, 200).concat("...") || "";
    const volumeAuthors = props.volumeResponse.volumeInfo.authors;
    const imageLinks = props.volumeResponse.volumeInfo.imageLinks;
    const volumeThumbnail = imageLinks ? imageLinks.thumbnail : "";

    const setVolumeInStore = () => {
        volumeStore.setVolume(props.volumeResponse);
    }

    return (
        <div data-testid="volumePreview" className="volumePreviewStyle">
            <Link
                to={{
                    pathname: "/browse/volume",
                }}
                onClick={setVolumeInStore}   
                style={{ textDecoration: 'none', color: "black" }}         
            >
                <Container fluid>
                    <Row >
                        <Col sm="12" md={{ size: 3, offset: 3 }} data-testid="volumePreviewThumbnail">
                            {volumeThumbnail ? <img src={volumeThumbnail} alt={`${volumeTitle} by authors ${volumeThumbnail}`} /> : null}
                        </Col>
                        <Col sm="12" md={{ size: 3 }}>
                            <Card>
                                <CardBody>
                                    <CardTitle data-testid="volumePreviewTitle">{volumeTitle}</CardTitle>
                                    <CardSubtitle tag="h6" className="mb-2 text-muted" data-testid="volumePreviewAuthors">{volumeAuthors}</CardSubtitle>
                                    <CardText data-testid="volumePreviewDescription">{volumeDescription}</CardText>
                                </CardBody> 
                            </Card>
                        </Col>
                    </Row>
                </Container>
            </Link>

        </div>
    );
}