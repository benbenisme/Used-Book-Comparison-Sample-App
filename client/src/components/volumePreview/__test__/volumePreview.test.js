import React from 'react';
import ReactDOM from 'react-dom';
import VolumePreview from "../VolumePreview";
import { BrowserRouter as Router } from "react-router-dom";
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

const exampleProps = {
    volumeResponse: {
        volumeInfo: {
            title: "test title",
            description: "test description",
            authors: "test author",
            imageLinks: {
            }
        }
    }
}

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><VolumePreview {...exampleProps} /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
});

test("renders volume preview corectly", () => {
    const { getByTestId } = render(<Router><VolumePreview {...exampleProps} /></Router>);
    getByTestId("volumePreview");
});

test("renders volume preview title corectly", () => {
    const { getByTestId } = render(<Router><VolumePreview {...exampleProps} /></Router>);
    expect(getByTestId('volumePreviewTitle')).toHaveTextContent(`${exampleProps.volumeResponse.volumeInfo.title}`);
});

test("renders volume preview authors corectly", () => {
    const { getByTestId } = render(<Router><VolumePreview {...exampleProps} /></Router>);
    expect(getByTestId('volumePreviewAuthors')).toHaveTextContent(`${exampleProps.volumeResponse.volumeInfo.authors}`);
});

test("renders volume preview dewcription corectly", () => {
    const { getByTestId } = render(<Router><VolumePreview {...exampleProps} /></Router>);
    expect(getByTestId('volumePreviewDescription')).toHaveTextContent(`${exampleProps.volumeResponse.volumeInfo.description}`);
});

it("matches snapshot", () => {
    const tree = renderer.create(<Router><VolumePreview {...exampleProps} /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
})