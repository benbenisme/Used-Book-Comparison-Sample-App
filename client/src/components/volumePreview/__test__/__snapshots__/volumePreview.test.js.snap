// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="volumePreviewStyle"
  data-testid="volumePreview"
>
  <a
    href="/volume"
    onClick={[Function]}
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-12 col-md-3 offset-md-3"
          data-testid="volumePreviewThumbnail"
        />
        <div
          className="col-sm-12 col-md-3"
        >
          <div
            className="card"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title"
                data-testid="volumePreviewTitle"
              >
                test title
              </div>
              <h6
                className="mb-2 text-muted card-subtitle"
                data-testid="volumePreviewAuthors"
              >
                test author
              </h6>
              <p
                className="card-text"
                data-testid="volumePreviewDescription"
              >
                test description
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
